JsonSchema/JsonSchemaSerializerContext.cs

```diff
@@ -0,0 +1,90 @@
+	[JsonSerializable(typeof(DependentSchemasKeyword))]
+	[JsonSerializable(typeof(DeprecatedKeyword))]
+	[JsonSerializable(typeof(DescriptionKeyword))]
+	[JsonSerializable(typeof(DynamicAnchorKeyword))]
+	[JsonSerializable(typeof(DynamicRefKeyword))]
+	[JsonSerializable(typeof(ElseKeyword))]
+	[JsonSerializable(typeof(EnumKeyword))]
+	[JsonSerializable(typeof(ExamplesKeyword))]
+	[JsonSerializable(typeof(ExclusiveMaximumKeyword))]
+	[JsonSerializable(typeof(ExclusiveMinimumKeyword))]
+	[JsonSerializable(typeof(FormatKeyword))]
+	[JsonSerializable(typeof(IdKeyword))]
+	[JsonSerializable(typeof(IfKeyword))]
+	[JsonSerializable(typeof(ItemsKeyword))]
+	[JsonSerializable(typeof(MaxContainsKeyword))]
+	[JsonSerializable(typeof(MaximumKeyword))]
+	[JsonSerializable(typeof(MaxItemsKeyword))]
+	[JsonSerializable(typeof(MaxLengthKeyword))]
+	[JsonSerializable(typeof(MaxPropertiesKeyword))]
+	[JsonSerializable(typeof(MinContainsKeyword))]
+	[JsonSerializable(typeof(MinimumKeyword))]
+	[JsonSerializable(typeof(MinItemsKeyword))]
+	[JsonSerializable(typeof(MinLengthKeyword))]
+	[JsonSerializable(typeof(MinPropertiesKeyword))]
+	[JsonSerializable(typeof(MultipleOfKeyword))]
+	[JsonSerializable(typeof(NotKeyword))]
+	[JsonSerializable(typeof(OneOfKeyword))]
+	[JsonSerializable(typeof(PatternKeyword))]
+	[JsonSerializable(typeof(PatternPropertiesKeyword))]
+	[JsonSerializable(typeof(PrefixItemsKeyword))]
+	[JsonSerializable(typeof(PropertiesKeyword))]
+	[JsonSerializable(typeof(PropertyDependenciesKeyword))]
+	[JsonSerializable(typeof(PropertyNamesKeyword))]
+	[JsonSerializable(typeof(ReadOnlyKeyword))]
+	[JsonSerializable(typeof(RecursiveAnchorKeyword))]
+	[JsonSerializable(typeof(RecursiveRefKeyword))]
+	[JsonSerializable(typeof(RefKeyword))]
+	[JsonSerializable(typeof(RequiredKeyword))]
+	[JsonSerializable(typeof(SchemaKeyword))]
+	[JsonSerializable(typeof(ThenKeyword))]
+	[JsonSerializable(typeof(TitleKeyword))]
+	[JsonSerializable(typeof(TypeKeyword))]
+	[JsonSerializable(typeof(UnevaluatedItemsKeyword))]
+	[JsonSerializable(typeof(UnevaluatedPropertiesKeyword))]
+	[JsonSerializable(typeof(UniqueItemsKeyword))]
+	[JsonSerializable(typeof(UnrecognizedKeyword))]
+	[JsonSerializable(typeof(VocabularyKeyword))]
+	[JsonSerializable(typeof(WriteOnlyKeyword))]
+	[JsonSerializable(typeof(JsonNode))]
+	[JsonSerializable(typeof(SchemaValueType))]
+	[JsonSerializable(typeof(string[]))]
+	[JsonSerializable(typeof(Dictionary<string, JsonSchema>))]
+	[JsonSerializable(typeof(Dictionary<string, bool>))]
+	[JsonSerializable(typeof(List<JsonSchema>))]
+	[JsonSerializable(typeof(List<string>))]
+	[JsonSerializable(typeof(JsonArray))]
+	[JsonSerializable(typeof(Dictionary<string, SchemaOrPropertyList>))]
+	[JsonSerializable(typeof(Dictionary<string, List<string>>))]
+	[JsonSerializable(typeof(int[]))]
+	[JsonSerializable(typeof(Dictionary<string, PropertyDependency>))]
+	internal partial class JsonSchemaSerializerContext : JsonSerializerContext
```

Comment:
We had talked about splitting this into two contexts, one for public types and one for internal.

Reply:
@eiriktsarpalis, what's the guidance for exposing serializer contexts?

Reply:
What Eirik told me is that only the TypeInfoResolver is needed so that callers can add it to their JsonSerializerOptions.TypeInfoResolverChain. I don't know if there's a pattern for this yet, but I've exposed JsonSchema.TypeInfoResolver public property to cover this.